### PR TITLE
RFC: implement degraded_on_stale

### PR DESCRIPTION
**Pull request by girst**
see discussion in #1883

### CODE REVIEW

**Comment by girst** [comment]
I've tacked on a second change (properly setting a timeout when opening the socket the first time)

**Comment by ultrabug** [comment]
I must admit that I got lost in all your work with @justyn on #1886 so I do not know if you reached an optimal yet or if I should wait for your signal guys

**Comment by girst** [comment]
On Sun, Jan 26, 2020 at 05:01:08AM -0800, Ultrabug wrote:
>I must admit that I got lost in all your work with @justyn on #1886 so I do not know if you reached an optimal yet or if I should wait for your signal guys

Yeah, that's a lengthy brainstorm. Essentially, we discovered a stall 
when connecting to captive portal networks, which I'm fairly confident 
(*knocks on wood*) 27b1991 fixes.

I'd like to see some confirmation about this from justyn, but otherwise 
I believe this has reached an optimum and is good to go.

**Comment by justyn** [comment]
> Yeah, that's a lengthy brainstorm. Essentially, we discovered a stall when connecting to captive portal networks, which I'm fairly confident (*knocks on wood*) [27b1991](https://github.com/ultrabug/py3status/commit/27b1991dd6d946c453cabd2b23bde90f97109a7f) fixes. I'd like to see some confirmation about this from justyn, but otherwise I believe this has reached an optimum and is good to go.

Sorry I couldn't reply properly before.

TLDR: yes this latest change seems to fix the problem! Great stuff @girst!

I did wonder if the two calls to setdefaulttimeout could be combined?

**Comment by girst** [comment]
On Sun, Jan 26, 2020 at 05:34:11AM -0800, Justyn Butler wrote:
>Sorry I couldn't reply properly before.

no worries!

>TLDR: yes this latest change seems to fix the problem! Great stuff @girst!

great, this gets the green light from me, then.

>I did wonder if the two calls to setdefaulttimeout could be combined?

I'd rather have them as close to the call to imaplib.IMAP{,_SSL} to make 
it clear that this sets the socket timeouts inside imaplib.

**Comment by ultrabug** [comment]
Hi guys, did you reach a consensus on what to do to move on plz?

**Comment by girst** [comment]
On Mon, Feb 03, 2020 at 11:42:22AM -0800, Ultrabug wrote:
>Hi guys, did you reach a consensus on what to do to move on plz?

that's good to go!

**Comment by ultrabug** [comment]
Thanks @girst
